### PR TITLE
Add flake8 checker action + fix flakes

### PR DESCRIPTION
**Pull request by yuvipanda**
- Copied .flake8 file from upstream JupyterHub, but removed
  the ignore for unused module imports and local variables
- Fix flake8 warnings
- Setup a github action for the lint

- Turn on flake8 warning for unused local variable
  and fix bug found due to this!

  Don't overwrite user's KUBECONFIG

  Without this, your local authentication with kubectl
  is reset to that of the serviceaccount used to deploy the
  cluster. If you try and switch cluster, it'll fail with
  strange errors.

  Now we properly set the env vars while authenticated so
  it does not leak out.

  Found because I turned on flake8 warning for unused
  local variables!

### CODE REVIEW

**Comment by choldgraf** [comment]
Could we just use a precommit check? We might wanna add things like black one day too and this could make it easier to extend with just a single job

**Comment by yuvipanda** [comment]
I've never managed to get pre-commit to work any time in any of my projects... For JupyterHub I manually run black :D Happy to take that if someone else can set it up tho.

This commit (https://github.com/2i2c-org/pilot-hubs/pull/411/commits/ea2c213865597dbed2bd19839517847637a82cd9) does fix a bug tho.

**Comment by choldgraf** [comment]
Yeah fair enough - maybe something we can just keep on our radar, don't think it should block this PR though since some lint is better than none 😅

File: .github/workflows/python-lint.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    paths:
+      - deployer/
+
+jobs:
+  flake8_py3:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Setup Python
+        uses: actions/setup-python@v1
```
Comment (GeorgianaElena) [inline .github/workflows/python-lint.yaml:14]: Is there any reason for not using the v2 version of the action? (I don't know how they differ in terms of features tho...)

Comment (yuvipanda) [reply .github/workflows/python-lint.yaml]: Ah good catch! I've updated :)

File: deployer/hub.py [hunk 2]
```diff
@@ -71,22 +68,28 @@ def auth_gcp(self):
         location = config.get('zone', config.get('region'))
         cluster = config['cluster']
         with tempfile.NamedTemporaryFile() as kubeconfig:
-            with decrypt_file(key_path) as decrypted_key_path:
+            orig_kubeconfig = os.environ.get('KUBECONFIG')
```
Comment (GeorgianaElena) [inline deployer/hub.py:71]: Oh wow... You caught this because KUBECONFIG wasn't used, right?
Hooray to having a linter now :tada:

**Review by GeorgianaElena — APPROVED** [review]
Thanks for setting this up, @yuvipanda!
I believe this is ready to go and hooray for having it prove its utility already! <3

File: deployer/hub.py (hunk 2 above)
Comment (yuvipanda) [reply deployer/hub.py:71]: EXACTLY! It was messing with my local credentials too

**Comment by yuvipanda** [comment]
Thanks, @GeorgianaElena!

**Comment by damianavila** [comment]
> Don't overwrite user's KUBECONFIG

That was a nice catch from the linter... and you!
Belated LGTM 😉

**Pull request merged**